xGame/Java/xgame/src/main/java/com/xgame/data/IUserRepository.java

```diff
@@ -8,4 +8,6 @@
 
 public interface IUserRepository extends JpaRepository<User, Integer> {
 	List<User> findByNicknameLikeAndIsDeletedFalseOrEmailAndIsDeletedFalse(String nickname, String email);
+
+	User findByEmailAndPasswordHashAndIsDeletedFalse(String email, String passwordHash);
 }
```

Comment:
return Optional<User> instead in case User cannot be found.

Reply:
Good call, changed